data_model/src/sorting.rs

```diff
@@ -0,0 +1,52 @@
+//! Structures and traits related to sorting.
+
+#[cfg(not(feature = "std"))]
+use alloc::{
+    string::{String, ToString as _},
+    vec::Vec,
+};
+
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "warp")]
+use warp::{Filter, Rejection};
+
+use crate::prelude::*;
+
+const SORT_BY_KEY: &str = "sort_by_metadata_key";
+
+/// Enum for sorting requests
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct Sorting {
+    /// Sort query result using [`Name`] of the key in [`Asset`]'s metadata.
+    pub sort_by_metadata_key: Option<Name>,
+}
+
+impl Sorting {
+    /// Creates a sorting by [`Name`] of the key.
+    pub fn by_metadata_key(key: Name) -> Self {
+        Self {
+            sort_by_metadata_key: Some(key),
+        }
+    }
+}
+
+impl From<Sorting> for Vec<(&'static str, String)> {
```

Comment:
what is the purpose of this?

Reply:
It's used in `prepare_query_request` to convert into parameters for warp [here](https://github.com/hyperledger/iroha/pull/2575/files#diff-f74337de83c9232a0cc949a8d6d7390cf6e46b011b79383d9666ede90a902526R623).